docs/sitemap_gen.py

```diff
@@ -543,26 +541,16 @@ def Validate(self, base_url, allow_fragment):
 
     # Test the lastmod
     if self.lastmod:
-      match = False
       self.lastmod = self.lastmod.upper()
-      for pattern in LASTMOD_PATTERNS:
-        match = pattern.match(self.lastmod)
-        if match:
-          break
-      if not match:
+      if not any(pattern.match(self.lastmod) for pattern in LASTMOD_PATTERNS):
         output.Warn('Lastmod "%s" does not appear to be in ISO8601 format on '
                     'URL: %s' % (self.lastmod, self.loc))
         self.lastmod = None
 
     # Test the changefreq
     if self.changefreq:
-      match = False
       self.changefreq = self.changefreq.lower()
-      for pattern in CHANGEFREQ_PATTERNS:
-        if self.changefreq == pattern:
-          match = True
-          break
-      if not match:
+      if all(self.changefreq != pattern for pattern in CHANGEFREQ_PATTERNS):
```

Comment:
Could this be `self.changefreq not in CHANGEFREQ_PATTERNS`?

Reply:
I see that the one above here, for `lastmod`, needs a `.match()` but this is just an `==`.

Another idea is to make it mirror the `lastmod` code with `not any()` instead of `all( ... != ...)`.

Reply:
+1 for `self.changefreq not in CHANGEFREQ_PATTERNS`.